src/components/DirectoryTree.tsx

```diff
@@ -303,16 +303,18 @@ export class DirectoryTree extends React.Component<DirectoryTreeProps, {
     });
   }
   lastClickedTime = Date.now();
+  lastClickedFile: File | null = null;
   onClickFile(file: File) {
     if (file instanceof Directory) {
       return;
     }
-    if (Date.now() - this.lastClickedTime < 1000 && this.props.onDoubleClickFile) {
+    if (Date.now() - this.lastClickedTime < 500 && this.lastClickedFile === file && this.props.onDoubleClickFile) {
```

Comment:
can move extract a constant, e.g. `const ThrottleClickTimeout = 500;` and place it at the line above? (it will be instead of comment)